db.go

```diff
@@ -1201,6 +1208,11 @@ func (db *DB) grow(sz int) error {
 	// Truncate and fsync to ensure file size metadata is flushed.
 	// https://github.com/boltdb/bolt/issues/284
 	if !db.NoGrowSync && !db.readOnly {
```

Comment:
What if `db.readOnly` is `false`, but `db.NoGrowSync` is `true`? My understanding is that the file size will still automatically increase if bbolt mmaps a size > fileSize and writes (WriteAt) data beyond the file real size. Unfortunately, we don't have test case to verify that.

I am thinking probably we don't need `NoGrowSync` at all, which was introduced in https://github.com/boltdb/bolt/pull/363. The original thought was to prevent preallocate disk space, but the implementation has changed a lot; we don't preallocate disk space for non-Windows platforms anymore; instead, we only grow the disk space when it's really needed in Commit. @benbjohnson @fuweid @tjungblu 

See also https://github.com/boltdb/bolt/pull/286

Reply:
I think if NoGrowSync doesn't get removed, then I should be doing the check outside of this if statement - ie., the MaxSize check should happen irrespective of NoGrowSync setting

Reply:
I think you still need to check `db.readOnly`, and only apply the max size check when it isn't readonly.